charts/application/templates/k8s-deployment.yaml

```diff
@@ -79,17 +85,35 @@ spec:
               protocol: {{ $ap.protocol }}
             {{- end }}
           livenessProbe:
+            {{- if .Values.livenessProbe.useCurl }}
+            exec:
+              command:
+                - curl
+                - -f
+                - -k
```

Comment:
`-k` (ignore untrusted certificates) isn't necessary for http urls.

but anyhow, as `curl` would need to be available in the app container, how about making it even more generic? having a value `probe.cmd` array? you could do the curl example in the value comments:
```
livenessProbe:
#   cmd: ['curl', '-f', 'http://localhost/call-me-plenty']
```

really like your new option, but when commands are used, it's not necessary http or the same port than the exposed one.